site/src/Librecores/ProjectRepoBundle/Controller/OrganizationController.php

```diff
@@ -121,198 +163,247 @@ public function settingsAction(Request $request, Organization $organization)
 
         return $this->render('LibrecoresProjectRepoBundle:Organization:settings.html.twig',
             array('organization' => $organization,
-                  'form' => $form->createView()));
+                  'form'         => $form->createView()));
     }
 
     /**
-     * Remove an organization
+     * Request to join an organization
      *
      * @param string $organizationName
      * @return Response
      */
-    public function removeAction($organizationName)
+    public function joinAction($organizationName)
     {
         $o = $this->getDoctrine()
                   ->getRepository('LibrecoresProjectRepoBundle:Organization')
                   ->findOneByName($organizationName);
 
         if (!$o) {
-            throw $this->createNotFoundException('No organization found with that name.');
+            throw $this->createNotFoundException(
+                'No organization found with that name.');
         }
 
-        // Remove the organization
-
-        if ($this->getUser() != $o->getOwner())
-            throw $this->createAccessDeniedException("You don't own this organization in order to remove it");
-
-        // TODO: Handle projects related to this organization!
+        $user = $this->getUser();
 
+        // Create new organization membership
+        $member = new OrganizationMember();
+        $member->setOrganization($o);
+        $member->setUser($user);
+        $member->setPermissions(OrganizationMember::PERMISSIONS_REQUEST);
         $em = $this->getDoctrine()->getManager();
-        $em->remove($o);
+        $em->persist($member);
         $em->flush();
 
-        return $this->render('LibrecoresProjectRepoBundle:Organization:remove.html.twig',
+        return $this->render('LibrecoresProjectRepoBundle:Organization:join.html.twig',
             array('organization' => $o));
     }
 
     /**
-     * Search for an organization
-     *
-     * @return Response
-     */
-    public function searchAction()
-    {
-        // Search for an organization
-
-        return $this->render('LibrecoresProjectRepoBundle:Organization:search.html.twig');
-    }
-
-    /**
-     * Request to join an organization
+     * Leave from an organization
      *
      * @param string $organizationName
      * @return Response
      */
-    public function joinAction($organizationName)
+    public function leaveAction($organizationName)
```

Comment:
Should we prevent leaving or allow it and set the organization to a default e.g 'librecores' creator? I don't have a preference, your call!